### PR TITLE
Logger: Fix double error messages/stack

### PR DESCRIPTION
**Pull request by ndelangen**
## What I did

Before:
Screenshot 2023-08-22 at 20 16 17

After:
Screenshot 2023-08-22 at 20 16 43

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

1. Break the build/dev somehow that would normally break storybook. I opted to breaking the manager builder.
   ```tsx
    export const executor = {
      get: async () => {
        const { build } = await import('esbuild');
        // return build;

        return undefined;
      },
    };
   ```
2. run any sandbox or the ui storybook
3. expect the error displayed in the console to only list the stack once!

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Comment by yannbf** [comment]
This is fantastic <3

On the matter of double messages, could you maybe take a look at webpack failures? They still log double error messages:
image

To reproduce:
1. create a webpack sandbox
2. add this to main config: `webpackFinal: (config) => ({})`

Totally fine if you don't want to do this or do it in a different PR

**Comment by ndelangen** [comment]
@yannbf I think this is a quick win already, Let's not have "Perfect" be the enemy of "Good".

File: code/lib/node-logger/src/index.ts [hunk 1]
```diff
@@ -25,13 +25,28 @@ export const logger = {
...
+      if (message instanceof Error && message.stack) {
+        msg = message.stack.toString();
+      } else {
+        msg = message.toString();
+      }
+
+      console.log(
+        msg
+          .replaceAll(process.cwd(), '.')
+          .replace(message.toString(), chalk.red(message.toString()))
```
Comment (yannbf) [inline code/lib/node-logger/src/index.ts:46]: You might want to swap them around, else message.toString() which could contain a file path, will not be the same as `msg` after the path is changed

```suggestion
          .replace(message.toString(), chalk.red(message.toString()))
          .replaceAll(process.cwd(), '.')
```

Comment (yannbf) [reply code/lib/node-logger/src/index.ts:46]: FYI this is what it looks like when the error message already contains ansi characters from chalk

image

**Pull request merged**